server.js

```diff
@@ -11,16 +11,76 @@ mongoose.Promise = Promise
 // overridden when starting the server. For example:
 //
 //   PORT=9000 npm start
-const port = process.env.PORT || 8080
+const port = process.env.PORT || 8000
 const app = express()
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
+//Model
+const Thought = mongoose.model("Thought", {
+  message: {
+    type: String,
+    required: true,
+    minlength: 5,
+    maxlength: 140
+  },
+  hearts: {
+    type: Number,
+    default: 0
+  },
+  name: {
+    type: String,
+    default: "anonymous",
+    minlength: 2,
+    maxlength: 50
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now
+  }
+})
+
 // Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
+app.get("/", async (req, res) => {
+  const thoughts = await Thought.find().sort({ createdAt: -1 }).limit(20).exec()
+  if (thoughts.length > 0) {
+    res.json(thoughts)
+  } else {
+    res.status(404).json({ message: "No happy thoughts" })
+  }
+})
+
+app.post("/", async (req, res) => {
+  //Get info sent by client to our API endpoint
+  const { message, name } = req.body
+  //Use our mongoose model to create a new database entry
+  const thought = new Thought({ message, name })
+  try {
+    //Sucess case
+    const savedThought = await thought.save()
+    res.status(201).json(savedThought)
+  }
+  catch (err) {
+    res.status(400).json({ message: "Couldn't save thought", error: err.errors })
+  }
+})
+
+app.post("/:thoughtID/like", async (req, res) => {
+  //Find specific thought
+  const thoughtID = req.params.thoughtID
```

Comment:
`const { thoughtID } = req.params` you can be consistent and do it this way here as well